comp/core/telemetry/telemetryimpl/telemetry_mock.go

```diff
@@ -58,12 +60,89 @@ func newMock(deps testDependencies) telemetry.Mock {
 	return telemetry
 }
 
+type internalMetric struct {
+	metric     *dto.Metric
+	metricType string
+}
+
+func (m *internalMetric) Labels() map[string]string {
+	labels := m.metric.GetLabel()
+	// labels are not necessarily in the order they were declared
+	// so we use a map to compare them
+	labelMap := make(map[string]string, len(labels))
+	for _, label := range labels {
+		labelMap[label.GetName()] = label.GetValue()
+	}
+	return labelMap
+}
+
+func (m *internalMetric) Value() float64 {
+	var value float64
+	switch m.metricType {
+	case "count":
+		value = m.metric.GetCounter().GetValue()
+	case "gauge":
+		value = m.metric.GetGauge().GetValue()
+	case "histogram":
+		value = m.metric.GetHistogram().GetSampleSum()
+	}
+
+	return value
+}
+
 func (t *telemetryImplMock) GetRegistry() *prometheus.Registry {
 	t.mutex.Lock()
 	defer t.mutex.Unlock()
 	return t.registry
 }
 
+func (t *telemetryImplMock) GetCountMetric(subsystem, name string) ([]telemetry.Metric, error) {
+	metricFamily, err := t.GetRegistry().Gather()
+	if err != nil {
+		return nil, err
+	}
+
+	return getMetric(metricFamily, "count", subsystem, name)
```

Comment:
Can you add a constant for "count" to make it explicit this is the same as the one line 82?